### PR TITLE
[WIP] Styling for filter search widgets

### PR DESCRIPTION
**Pull request by mazameli**
**Things I can do**
- [x] Update grays
- [x] Token colors
- [x] Input border color 
- [x] Input and token padding/spacing
- [x] Border below table+field name
- [x] Make keyboard highlight state the same as the `hover` state
- [x] Make list item highlight state bg color expand full width
- [x] Dial in font sizes
- [x] Fix margin-bottom of tokens when they wrap to subsequent rows
- [x] Fix the placeholder text padding
- [x] Swap out existing magnifying glass icon in `icon_paths.js` with the new one (@kdoh, I ended up creating one identical to our existing `zoom` icon, without the plus symbol, since it was almost exactly like the one in my Sketch file, but a little chunkier and with a rounded handled, which seemed to fit better with our other icons.)
- [x] Dial in drop-shadows for dashboard/SQL filters
- [ ] Search default/empty state needs the magnifying glass asset and larger popover height

**Things I could use help with**
- [x] Hide all operators under an Options button. (Had considered putting this in a dropdown instead, a la the new Date filter, but it had some gnarly UI design implications for some of the possible states.)
- [ ] Get dashboard and SQL filter popovers to have the same sizing as query builder ones, along with the same primary button placement and size (bottom-right of popover, not full-width)
- [ ] Make it so that the token box only has scroll bars when necessary (I'm seeing that same ever-present scrollbars problem that we had recently with the dashboard text cards.)
- [ ] Fixed width and min/max-height height of the popover — not sure I know how to conditionally set a fixed width for this type of filter without without mucking up how other filters, like Date ones, depend on the popover having flexible width and height. (Max-height is probably less of a problem.) The goal is to keep the width from changing so much as additional values are added to `TokenField`.
- [x] Text for case when all possible filter values have been selected (low-prio)
- [x] Max height for token input box + overflow:scroll — thought I was doing this right, but having trouble figuring out how to get my cxs variable to play nice with the cx function that was already being used to populate the classNames for the input.
- [x] Increase input border thickness — I looked at `RecipientPicker.jsx` used in the Pulse edit form to try and figure out where the thicker borders are coming from, but couldn't figure it out.
- [x] Text for "no results" state
- [x] Put `Case sensitive` checkbox 
- [x] If `Is empty` or `Not empty` is selected, make the search box et al go away

### CODE REVIEW

File: frontend/src/metabase/components/TokenField.jsx [hunk 1]
```diff
@@ -383,19 +389,19 @@ export default class TokenField extends Component {
 
         const valuesList =
           <ul
-              className={cx("m1 px1 pb1 bordered rounded flex flex-wrap bg-white", {
-                [`input--focus border-${color}`]: this.state.focused
+              className={cx("m1 p0 bordered rounded flex flex-wrap bg-white", {inputBoxClasses}, {
```
Comment (tlrobinson) [inline frontend/src/metabase/components/TokenField.jsx]: I don't think you want the brackets around `inputBoxClasses`. That's going to set a class called "inputBoxClasses" to the value of inputBoxClasses

File: frontend/src/metabase/css/core/colors.css [hunk 2]
```diff
@@ -3,7 +3,8 @@
   --brand-light-color: #CDE3F8;
   --brand-saturated-color: #2D86D4;
 
-  --base-grey: #f8f8f8;
+  --base-grey: #F8F9FA;
+  --grey-5percent: color(var(--base-grey) shade(5%));
```
Comment (tlrobinson) [inline frontend/src/metabase/css/core/colors.css]: I don't really like that name but I don't know what's better.

**Comment by tlrobinson** [comment]
> Put Case sensitive checkbox

This is already present, but only for filters that support it (contains, etc)

> If Is empty or Not empty is selected, make the search box et al go away

This is the current behavior, no?

**Comment by tlrobinson** [comment]
I pushed a commit that fixes a few of these things.

File: frontend/src/metabase/css/core/colors.css (hunk 2 above)
Comment (mazameli) [reply frontend/src/metabase/css/core/colors.css]: I know, it sucks.

**Comment by mazameli** [comment]
>> Put in case-sensitive checkbox
> This is already present, but only for filters that support it (contains, etc)

Gotcha. 👍 

>> If `Is empty` or `Not empty` is selected, make the search box et al go away
> This is the current behavior, no?

True that. I wrote that while looking through the designs rather than the existing branch, so mea culpa.

File: frontend/src/metabase/css/core/colors.css (hunk 2 above)
Comment (tlrobinson) [reply frontend/src/metabase/css/core/colors.css]: `grey-05`?

**Comment by tlrobinson** [comment]
@mazameli Any reason not to just merge this into the main PR?

File: frontend/src/metabase/css/core/colors.css (hunk 2 above)
Comment (kdoh) [reply frontend/src/metabase/css/core/colors.css]: `grey-05` seems like it might conform more to our existing setup.

**Comment by mazameli** [comment]
@tlrobinson No, especially now that I've done most or all of what I'm capable of doing, so the chances of me breaking the main PR are much diminished. :)

**Comment by tlrobinson** [comment]
Merged with #4241

**Pull request closed without merging**